c++/src/kj/common.h

```diff
@@ -2194,12 +2194,32 @@ inline void memzero(T& t) {
   memset(&t, 0, sizeof(T));
 }
 
+namespace _ {
+template <size_t N>
+struct ByteLiteral {
+  // Helper class to implement _kjb suffix: constexpr is not allowed to cast `char*`
+  // to `unsigned char*`. To overcome this limitation every char needs to be cast individually
+  // (which is apparently ok for constexpr).
+  // This class may not contain any private members, or else you get a
+  // misleading compiler error.
+
+  constexpr ByteLiteral(const char (&init)[N]) {
+    for (auto i = 0; i < N; ++i) {      
+      data[i] = (kj::byte)(init[i]);
+    }
+  }
+  constexpr size_t size() const { return N - 1; }
+  kj::byte data[N];
```

Comment:
As long as we're here, how about making the size `N - 1` so that we don't store the NUL terminator?